packages/next-swc/crates/core/src/next_dynamic.rs

```diff
@@ -61,8 +61,14 @@ impl Fold for NextDynamicPatcher {
     fn fold_call_expr(&mut self, expr: CallExpr) -> CallExpr {
         if self.is_next_dynamic_first_arg {
             if let Callee::Import(..) = &expr.callee {
-                if let Expr::Lit(Lit::Str(Str { value, .. })) = &*expr.args[0].expr {
-                    self.dynamically_imported_specifier = Some(value.to_string());
+                match &*expr.args[0].expr {
+                    Expr::Lit(Lit::Str(Str { value, .. })) => {
+                        self.dynamically_imported_specifier = Some(value.to_string());
+                    }
+                    Expr::Tpl(Tpl { exprs, quasis, .. }) if exprs.len() == 0 => {
+                        self.dynamically_imported_specifier = Some(quasis[0].raw.to_string());
+                    }
```

Comment:
@magic-akari  Is it possible to preserve the original specifier?

E.g.:

```ts
const componentRoot = '@/some-components'
const Component1 = dynamic(() => import(`${componentRoot}/component1`));
const Component2 = dynamic(() => import(`${componentRoot}/component2`));
```

.. should turn into:

```ts
const componentRoot = '@/some-components'
[require.resolveWeak(`${componentRoot}/component1`]
[require.resolveWeak(`${componentRoot}/component2`]
```

Note that it is up to Webpack to handle, evaluate and resolve `require.resolveWeak`.

Reply:
Yes, that's how it works now.
I've updated the test case.

Reply:
`exprs.len() == 0` means that we only handle template literal if there is no embedded expressions.
Any template literals that contain embedded expressions will be ignored.

Example

```JavaScript
import dynamic from 'next/dynamic'

const DynamicComponent = dynamic(() => import(`../components/hello`))

const componentRoot = '@/some-components'
const Component1 = dynamic(() => import(`${componentRoot}/component1`));
const Component2 = dynamic(() => import(`${componentRoot}/component2`));
```

output

```JavaScript
import dynamic from 'next/dynamic';
const DynamicComponent = dynamic(()=>import(`../components/hello`), {
    loadableGenerated: {
        webpack: ()=>[
                require.resolveWeak("../components/hello")
            ]
    }
});
const componentRoot = '@/some-components';
const Component1 = dynamic(()=>import(`${componentRoot}/component1`));
const Component2 = dynamic(()=>import(`${componentRoot}/component2`));
```

Is this the expected result?
@SukkaW